### PR TITLE
Create merge comments action

### LINKED ISSUE #7880: Combining the bot comments into only one comment
## Is your feature request related to a problem? Please describe.
At the moment, we have 4 actions that adding comments to the PR page:
- Generate ZIP file.
- TypeScript Errors Report
- Compress Size Action
- Script Dependencies Report

While they're all helpful information, but at the same time, having each of them add comments to the PR page creates some issues/annoyances:
- They trigger new notifications after PR is created, contributors check the PR and realized it's just bot comments, not a review or comments from other contributors. 
- Those comments make the PR harder to read. It's hard to follow the conversation because bot comments in between. The comments also make the page longer, which costs some page scrolls to find the conversation.
- There are many factors affecting the time when the workflow finished running, which makes the order of bot comments unpredictable (this is just a minor issue btw).

## Describe the solution you'd like
By combining all bot comments into one only one comment, we can:
- Reduce the notification of newly created PRs. It doesn't solve the notification issue entirely though.
- Make the PR page shorter and easier to read.
- Make the helpful information created by GH Actions easier to look up as everything is in the same place.
- Make the PR page scalable, by combining all bot comments into one, we can add more integrations/actions in the future without bloating the PR page.

We'll need an action (which may or may not exist on the marketplace) that acts as the commenter. Instead of posting comments directly, all workflows should use that action to create/edit the comment.

We'll also need to improve the content of each comment to make them more compact while still containing enough information.

## Describe alternatives you've considered
We can edit the PR description directly but that direction is problematic:
- Contributor can accidentally remove content created by bot.
- Unexpected results if the bot updates the description while the contributor is still editing.

## Additional context
Example combined comment (which can be more compact)

image

### PR DESCRIPTION
**Pull request by dinhtungdu**
Fixes #7880

This PR:
- Creates a new action that acts as a commenter: it takes the output of various actions, then posts them to only one comment on the PR page.
- Updates existing action to set the comment content as an action output.
- Updates existing workflows to use the new action.

### Screenshots

image

image

Or see https://github.com/woocommerce/woocommerce-blocks/pull/7973#issuecomment-1359252211

### CODE REVIEW

File: .github/comments-aggregator/action.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: 'Merge Comments'
+description: 'Merge bot comment into one comment to keep PR page clean.'
```
Comment (Aljullu) [inline .github/comments-aggregator/action.yml]: Nit: should this be plural?
```suggestion
description: 'Merge bot comments into one comment to keep PR page clean.'
```

File: .github/comments-aggregator/action.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        description: 'GitHub token'
+        required: true
+    section-id:
+        description: 'Comment section ID for the action to know which part to update.'
+        required: true
+    content:
+        description: 'Comment content.'
+        required: true
+    order:
+        description: 'Order of the comment'
```
Comment (Aljullu) [inline .github/comments-aggregator/action.yml:14]: Nit: I think we should either add a period at the end of each description or not. Currently it's inconsistent: `Comment content.` (has period) and `Order of the comment` (has no period).

File: .github/comments-aggregator/utils.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
+const indentifier = `<!-- comments-aggregator -->`;
```
Comment (Aljullu) [inline .github/comments-aggregator/utils.js]: Typo: `identifier`.

**Review by Aljullu — APPROVED** [review]
I really liked the way you implemented this, @dinhtungdu! And very happy to get this merged to reduce the comments noise.

I left some minor comments, but pre-approving.

**Comment by dinhtungdu** [comment]
@Aljullu I addressed your review, thank you so much! I also added a README file to make using this action easier. Because of that, can you please take another look?

File: .github/comments-aggregator/README.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
+# comments-aggregator
+
+> This GitHub Action helps you keep the PR page clean by merging comments/reports
+> by multiple workflows into a single comment.
+
+![screenshot](./screenshot.png)
+
+## Usage
+
+This action is meant is be used as the poster/commenter. Instead of having
```
Comment (Aljullu) [inline .github/comments-aggregator/README.md]: `This action is meant is be used` → `This action is meant to be used`

File: .github/comments-aggregator/README.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+                  repo-token: '${{ secrets.GITHUB_TOKEN }}'
+                  compare: assets-list/assets.json
+                  create-comment: false
+            - name: Append report
+              uses: ./.github/comments-aggregator
+              with:
+                  repo-token: '${{ secrets.GITHUB_TOKEN }}'
+                  section-id: compare-assets-with-trunk
+                  content: ${{steps.compare-assets.outputs.comment}}
+```
```
Comment (Aljullu) [inline .github/comments-aggregator/README.md]: Nit: I would move this code block one indentation level up, to match the code block below.

**Review by Aljullu — APPROVED** [review]
Two small comments, but LGTM.

**Comment by fabiankaegy** [comment]
Hey @dinhtungdu 👋 

Do you think the workflow that adds any changes in dependencies could ever be made publically available? That would be super useful for anyone building blocks :)

**Comment by dinhtungdu** [comment]
Heyy @fabiankaegy,

> Do you think the workflow that adds any changes in dependencies could ever be made publically available? That would be super useful for anyone building blocks :)

If the action can help others, I'm 💯 agree with you.

@mikejolley as you're the author of the `Compare Assets` action, what do you think about publishing it as a dedicated action?

cc @nerrad

**Comment by nerrad** [comment]
Good candidate for https://github.com/woocommerce/automations?

**Comment by fabiankaegy** [comment]
After looking at the implementation here a bit more I think the version here is a little to purpose build for this repository setup and it doesn't use the `.asset.php` files which would be the case for most simple plugins. 

But seeing this inspired me to pick up work on an older exploration I had done that tries to do the same thing. I published it here: https://github.com/fabiankaegy/monitor-wordpress-dependencies-action
